### PR TITLE
[extension] command.uivoid

### PR DESCRIPTION
**Pull request by corrideluca**
## Summary

Implements the `command.uivoid` extension proposed and confirmed in #2997.

- `create`, `credentials`, `oauth-config`, and `login` are reviewed: they mutate remote account/project state (or, for `login`, write a local session file every later command trusts).
- `skill --install` is reviewed: it writes a file a Codex agent loads automatically in later sessions.
- `whoami`, `prompt`, `logout`, and bare `skill` (no `--install`) stay auto-allowed — pure reads or fully local/reversible.
- `--help`/`-h` are genuine safe variants on every rule, since commander.js intercepts them before the action handler runs, even alongside an otherwise-reviewed flag (e.g. `uivoid skill --install --help` never installs anything).
- Covers the direct `uivoid` binary plus common npm launcher wrappers: `npx`/`bunx uivoid`, `pnpm`/`yarn uivoid`, `npm`/`pnpm exec uivoid`, `pnpm`/`yarn dlx uivoid`.

External, opt-in, mirrors `command.repo2nb`'s structure. Registered in the three places that turned out to matter (traced by reading the actual runtime, not guessed):
1. `command_builtin_extension_catalog.py` — the direct extension catalog.
2. `contracts/extensions/trust-class-map.v1.json` — the curated map that actually gates default enablement (the contribution manifest's own `trustClass` field is validated but not what `extension_trust.py` consults).
3. `docs/guard/extensions/catalog.v1.json` — regenerated via `scripts/export_extension_directory.py`.

## Test plan

- [x] `pytest tests/test_guard_command_uivoid_extensions.py` — review cases, npm launcher wrapper forms, inert-until-enabled, safe/help commands, reference URL check
- [x] `pytest tests/test_guard_extension_trust.py tests/test_guard_extension_contribution.py tests/test_guard_extension_directory_export.py tests/test_guard_mcp_server_contribution.py` — confirms the extension is correctly wired into the trust map, contribution schema validation, and generated directory export
- [x] `ruff check` / `ruff format --check` on all changed files

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added support for the `uivoid` command extension across common package runners.
  * Added review safeguards for project creation, credential handling, OAuth configuration, login, and skill installation commands.
  * Added safer alternatives and help guidance for potentially risky operations.
  * Classified the extension as external and opt-in, remaining inactive until enabled.

* **Documentation**
  * Documented the `uivoid` extension and its protected operations.

* **Tests**
  * Added coverage for command variants, safe commands, trust classification, and protection of secret values.

### CODE REVIEW

**Comment by corrideluca** [comment]
Addressed both of Qodo's findings in ab2e350:

- **Launcher-option bypass (real bug):** the npx/bunx/pnpm/yarn wrapper matchers required `uivoid` to be the exact first token after the launcher, so `npx --yes uivoid create app`, `pnpm --silent uivoid create app`, etc. shifted the subcommand prefix and skipped review. Fixed by declaring those launcher flags as interspersed (same option/flag sets already reviewed for the `supabase` extension). Verified the new test cases fail without the fix and pass with it.
- **Test coverage gap:** added `--exclude-destructive`, `--flag=value` assignment syntax, the launcher-bypass cases above, and an explicit assertion that evidence for secret-bearing commands (`--auth-key`, `--token`, `--client-secret`) never contains the matched secret value.

47 targeted tests pass (`test_guard_command_uivoid_extensions.py` plus the trust map, contribution schema, directory export, and MCP contribution suites), `ruff check` and `ruff format --check` clean.

**Review by kantorcodes — DISMISSED** [review]
Please wire this through the current installed-runtime contract before we run the fork CI. `command.uivoid` is absent from `command_action_risk_classes.py`, the wheel `force-include`, frozen staging map, and `contracts/extensions/native-command-program.v1.json`; the generated README, control baseline, decision-diff, and projection artifacts also were not refreshed. Regenerate with the repo scripts and add installed wheel/frozen coverage so packaged behavior and contribution metadata match source.

**Comment by corrideluca** [comment]
Wired through in 5cdfef8:

- **Regenerated via the repo's own scripts** (all confirmed stale before, clean `--check` after): `contracts/extensions/native-command-program.v1.json`, `docs/guard/extensions/README.md`, `tests/fixtures/guard-command-corpus/decision-diff-report.json`.
- **Patched using the same library calls the tests themselves use to compute expected values**: `tests/fixtures/extension-controls/catalog-baseline.v1.json` (merged in the new permission/rule ids, left every pre-existing key's order untouched rather than doing a full re-sort that would've touched unrelated extensions), and `contracts/managed-controls/v1/extension-projection-digest-vector.json` + the matching digest constants in `tests/test_policy_bundle_delivery_runtime.py`.
- **Added the two missing per-extension registrations** — `command.uivoid.json` was absent from both the wheel `force-include` in `pyproject.toml` and the frozen staging map in `scripts/release/stage_guard_cloud_review_artifacts.py`, unlike every other merged external extension.

One I checked and left alone: `command_action_risk_classes.py`. That's a legacy lookup for action classes that predate rules declaring `risk_classes` directly — an unmapped action class returns `()`, which is always a valid subset of the extension's own declared risk classes (verified both relevant tests pass). `command.repo2nb` itself has no entry there either, so adding one for uivoid would deviate from the pattern rather than match it.

One I could not verify and want to flag rather than guess at: `rust/crates/guard-command/tests/fixtures/native-command-observations-v1.json` does look registry-derived (73/546 cases reference repo2nb), and `scripts/generate_native_command_fixtures.py` would regenerate it, but that script hard-requires CPython 3.12 exactly (`"Native v1 oracle requires CPython 3.12 / UCD15"`) and I don't have that pinned interpreter or `cargo`/`rustc` in this environment to regenerate or verify it. Left it for the Rust authority CI check to confirm rather than committing something unverifiable.

187 targeted tests pass across all affected suites; `ruff check`/`ruff format --check` clean.

File: src/codex_plugin_scanner/guard/runtime/command_uivoid_extensions.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+
+def _uivoid_matchers(
+    *subcommands: str,
+    required_flags: frozenset[str] = frozenset(),
+) -> tuple[ExecutableMatcher, ...]:
+    return (
+        executable_matcher("uivoid", *subcommands, required_flags=required_flags),
+        ExecutableMatcher(
+            executables=executable_names("npx") | executable_names("bunx"),
+            subcommands=("uivoid", *subcommands),
```
[the remark this answers is not in the document]
Comment (corrideluca) [reply src/codex_plugin_scanner/guard/runtime/command_uivoid_extensions.py]: Confirmed real — `npx uivoid@latest create` and `npm exec uivoid@1.2.3 create` evade every wrapper rule, since `ExecutableMatcher`'s subcommand comparison is an exact-token match, not a prefix.

I tried fixing it with a small custom matcher (package name with an optional `@version`/`@tag` suffix, checked before the fixed subcommand token). It works for the Python-side rule evaluation, but `scripts/build_native_command_program.py` hard-fails on it: `native_command_program.py` keeps a deliberately closed allowlist of matcher types (`"Keep this an explicit import/identity allowlist... would admit unreviewed Python semantics"`) — even `repo2nb`'s own custom matcher (`Repo2nbUnresolvedExpansionMatcher`) had to be explicitly added there by name. Self-admitting a brand-new matcher type into that list from an external contribution defeats the point of it being a reviewed allowlist, and I checked — none of the existing whitelisted primitives (`ExecutableMatcher`, `ArgumentMatcher`, `SubcommandOperandPrefixMatcher`) fit a variable-suffixed token in this position followed by a fixed subcommand.

So in 258a97e I left this specific gap unfixed and documented in the test file rather than route around it with an unreviewed matcher, and added the two cases your finding correctly noted already work (`--` before the package name — the matcher already consumes it, both `npx -- uivoid create` and `npm exec -- uivoid create` pass).

If this is worth closing, it'd need either a new reviewed matcher primitive added to `native_command_program.py`'s allowlist (useful generally — probably not just uivoid runs into "npx pkg@latest") or maintainer guidance on an existing primitive I'm missing. Happy to implement either if you point me at the right shape.

**Review by kantorcodes — DISMISSED** [review]
Two blockers remain on this head. Map every UIVoid action class in `command_action_risk_classes.py`; runtime evaluation calls `risk_classes_for_command_action`, so those classes currently publish no mapped risks. Also close the `uivoid@<version|tag>` launcher bypass with a reviewed matcher/native op and Python/Rust parity tests for `npx` and `npm exec` forms.

**Review by kantorcodes — COMMENTED** [review]
Nice, the risk mapping and versioned package matching are fixed. `main` has moved three commits since this branch; could you rebase onto `42f303d` and regenerate the source-bound Extension artifacts before we run fork CI?

**Comment by corrideluca** [comment]
Both blockers are fixed and pushed (risk-class mapping, and the `uivoid@<version|tag>` bypass via a new `VersionedPackageSubcommandMatcher` primitive with Python/Rust parity tests — also closed a follow-up bare-`pkg@` edge case gitar-bot caught on the same matcher).

While rebasing onto `42f303d` I hit something bigger than a routine conflict: `main` has since moved further (`#3003`, "declarative-native-authoring") and it looks like it replaces the whole compilation path this PR is built on:

- `src/codex_plugin_scanner/guard/runtime/native_command_program.py` — the Python compiler with the `_reviewed_types()` allowlist I extended for the new matcher — is deleted entirely.
- The native command program is now compiled by a new Rust binary (`guard-command-source compile`) from canonical JSON sources under `contributions/command-sources/*.json`, not from the Python `CommandSafetyRule`/`ExecutableMatcher` classes.
- That migration looks partial: I checked the current compiled `native-command-program.v1.json` on `main` and `command.errand.*` rules (merged via #2899, after this branch first synced) aren't in it at all — no `command.errand.json` source file exists yet, so Errand isn't enforced by the native path right now.

Given that, `command_uivoid_extensions.py` is written against a compiler that no longer exists on `main`, and porting it to the new JSON-source format would mean inventing an equivalent for the versioned-package matcher in a schema I haven't seen an example of yet (nothing in `contributions/command-sources/*.json` needs that shape currently).

Before I sink more time into either direction — how do you want new extension PRs like this one to proceed during the migration? Land uivoid as-is (old path) and let it get ported in a follow-up alongside Errand, or is there a reference/example for the new source format I should be authoring against directly?

**Review by kantorcodes — DISMISSED** [review]
Nice, the earlier matcher fixes are still intact. Current `main` now builds Extension behavior from `contributions/command-sources/`, and this branch still carries the older Python/native-program wiring. Could you refresh from `e8c2e21` and port `command.uivoid` to the declarative source format, keeping the versioned package cases as behavior fixtures?

**Review by kantorcodes — DISMISSED** [review]
The builder is failing because `docs/guard/extensions/catalog.v2.json` is stale: the regenerated v2 directory includes `command.uivoid`, but that file is unchanged in this PR. Could you run `python scripts/export_extension_directory.py`, commit the updated v2 catalog, and let the builder rerun?

**Review by kantorcodes — CHANGES_REQUESTED** [review]
The catalog refresh is fixed. The builder now rejects `tests/fixtures/command-source-uivoid.v1.json` before its cases run because it lacks the current `guard.command-extension-fixtures.v1` build envelope. Could you regenerate that fixture so `build.sources` contains the canonical `contributions/command-sources/command.uivoid.json` source, then rerun the Extension builder?

**Review by kantorcodes — DISMISSED** [review]
`tests/fixtures/guard-command-corpus/decision-diff-report.json` is stale on this head; Desktop contract fails at `tests/guard_command_decision_diff.py --check`. Please regenerate it with `uv run --no-sync python tests/guard_command_decision_diff.py --write`, review the fixture, then rerun with `--check`.

**Review by kantorcodes — APPROVED** [review]
